main.cpp

```diff
@@ -126,384 +167,529 @@ struct CarWash
 };
-/*
-Thing 7) Magazine
-5 properties:
-    1) Maximum Capacity (int)
-    2) Type (extended, drum, standard) (std::string)
-    3) Material (plastic, metal) (std::string)
-    4) Caliber (float)
-    5) Current amount of bullets left (int)
-3 things it can do:
-    1) Store rounds for pewpew
-    2) Feed rounds into feeding system
-    3) Prevent loading of incorrect rounds
- */
+
+bool Scope::provideSightPicture(float targetSize, int distance) 
+{
+    bool isVisible;
+    float calc = targetSize/distance;
+    if (calc > 100.0f)
+    {
+        isVisible = true;
+    }
+    else
+    {
+        isVisible = false;
+    }
+    return isVisible;
+}
+
+bool Scope::illuminateSightPicture(int scopeBatteryCapacity, int environmentLightLevel)
+{
+    bool isIlluminating;
+    if (scopeBatteryCapacity > 0 || environmentLightLevel > 0)
+    {
+        isIlluminating = true;
+    }
+    else
+    {
+        isIlluminating = false;
+    }
+    return isIlluminating;
+}
+
+bool Scope::alignmentBarrelToTarget(int elevationBarrel, int elevationTarget, int distance, float caliber)
+{
+    bool isAligned;
+    if (caliber * (elevationBarrel - elevationTarget)/distance > 50)
+    {
+        isAligned = true;
+    }
+    else
+    {
+        isAligned = false;
+    }
+    return isAligned;
+}
+
 struct Magazine
 {
-    // Maximum Capacity (int)
     int maxCapacity = 30;
-    // Type (extended, drum, standard) (std::string)
     std::string type = "standard";
-    // Material (plastic, metal) (std::string)
     std::string material = "plastic";
-    // Caliber (float)
     float caliber = 5.56f;
-    // Current amount of bullets left (int)
     int currentCapacity = 30;
 
-    // Store rounds for pewpew
-    int holdRounds(int roundsLoaded, int roundsExpended); // Returns number of current rounds in capa
-    // Feed rounds into feeding system
+    int holdRounds(int roundsLoaded, int roundsExpended); 
     int feedRounds(bool isChamberJammed);
-    // Prevent loading of incorrect rounds
     void filterIncorrectRounds(bool isCaliberCorrect); 
 };
-/*
-Thing 8) Bullet
-5 properties:
-    1) Caliber (float)
-    2) Powder load (int)
-    3) Powder brand (std::string)
-    4) Bullet weight (grains) (float)
-    5) Bullet type (HP, AP, FMJ, HE) (std::string)
-3 things it can do:
-    1) Engage rifling in barrel
-    2) Accelerate when fired
-    3) Impact target downrange
- */
+
+int Magazine::holdRounds(int roundsLoaded, int roundsExpended)
+{
+    int roundsLeft = roundsLoaded - roundsExpended;
+    return roundsLeft;
+}
+
+int Magazine::feedRounds(bool isChamberJammed)
+{
+    int roundInChamber;
+    if (isChamberJammed == true)
+    {
+        roundInChamber = 0;
+    }
+    else
+    {
+        roundInChamber = 1;
+    }
+    return roundInChamber;
+}
+
+void Magazine::filterIncorrectRounds(bool isCaliberCorrect)
+{
+    if (isCaliberCorrect == false)
+    {
+        std::cout << "Incorrect caliber loaded! Unable to fire.\n";
+    }
+    return;
+}
```

Comment:
Functions that don't return anything don't need a `return;` statement
- [ ] revise